src/network/protocol/redis/command/network_protocol_redis_command_mget.c

```diff
@@ -0,0 +1,316 @@
+/**
+ * Copyright (C) 2018-2022 Vito Castellano
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the BSD license.  See the LICENSE file for details.
+ **/
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <strings.h>
+#include <arpa/inet.h>
+#include <assert.h>
+
+#include "misc.h"
+#include "exttypes.h"
+#include "log/log.h"
+#include "clock.h"
+#include "spinlock.h"
+#include "data_structures/small_circular_queue/small_circular_queue.h"
+#include "data_structures/double_linked_list/double_linked_list.h"
+#include "data_structures/queue_mpmc/queue_mpmc.h"
+#include "slab_allocator.h"
+#include "data_structures/hashtable/mcmp/hashtable.h"
+#include "data_structures/hashtable/mcmp/hashtable_op_get.h"
+#include "protocol/redis/protocol_redis.h"
+#include "protocol/redis/protocol_redis_reader.h"
+#include "protocol/redis/protocol_redis_writer.h"
+#include "network/protocol/network_protocol.h"
+#include "network/io/network_io_common.h"
+#include "config.h"
+#include "fiber.h"
+#include "network/channel/network_channel.h"
+#include "storage/io/storage_io_common.h"
+#include "storage/channel/storage_channel.h"
+#include "storage/db/storage_db.h"
+#include "network/protocol/redis/network_protocol_redis.h"
+#include "network/network.h"
+#include "worker/worker_stats.h"
+#include "worker/worker_context.h"
+
+#define TAG "network_protocol_redis_command_mget"
+
+struct mget_command_key_data {
+    char *key;
+    size_t key_length;
+    size_t key_offset;
```

Comment:
A tiny tiny and optional improvement would be to move `key_offset` under `mget_command_context` because it's need only while receiving the data and only one at time, we don't need to preserve it at all.

Or we can potentially even use just `key_length` .

The reason is that if someone requests 100000 keys the memory needed for the struct above will be 24 bytes so 2.4MB will be required, if we drop the key_offset from there moving it to `mget_command_context` (or use key_length in its place) it will only need 16 bytes and therefore 1.6MB.